src/main/webapp/app/entities/programming-exercise/instructions/programming-exercise-editable-instruction.component.ts

```diff
@@ -115,6 +117,8 @@ export class ProgrammingExerciseEditableInstructionComponent implements AfterVie
                     target.style.height = event.rect.height + 'px';
                 }
             });
+
+        this.templateChanged.subscribe(() => this.generateHtml());
```

Comment:
There are 2 thinks I don't like about templateChanged here:
- It is too specific to the fact that the template changed - the editable instructions shouldn't even know about the fact that there are templates. It should be better named `instructionsChanged`
- More important: This could also be a normal Input that is detected in `NgOnChanges` as far as I see (we have a util problemStatementHasChanged). I would only use passed observables in very rare cases as it makes the control flow harder to evaluate.

Reply:
Great suggestion! I now use ngOnChanges and the util for detecting language changes and re-rendering the preview html